### PR TITLE
Add docs for epilog() and empty() recommendations

### PR DESCRIPTION
**Pull request by markstory**
*  Document using locking statements with epilog() #3981
*  Add section on avoiding empty(). cakephp/cakephp#8397
* Reorder query builder post-processing section and wrap long lines.

### CODE REVIEW

File: en/contributing/cakephp-coding-conventions.rst [hunk 1]
```diff
@@ -544,6 +544,72 @@ underscore character, for example::
 
     define('LONG_NAMED_CONSTANT', 2);
 
+Avoid empty()
```
Comment (ionas) [inline en/contributing/cakephp-coding-conventions.rst]: Careful usage of empty() / isset()
=========================

**Comment by chinpei215** [comment]
@markstory Thanks!

File: en/contributing/cakephp-coding-conventions.rst (hunk 1 above)
Comment (markstory) [inline en/contributing/cakephp-coding-conventions.rst]: Done.

**Comment by markstory** [comment]
Thanks @ravage84

**Pull request merged**